### PR TITLE
fix: Optimizebatchnorm

### PR DESCRIPTION
**Pull request by hello-fri-end**

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy/functional/ivy/experimental/norms.py [hunk 1]
```diff
@@ -195,20 +195,28 @@ def batch_norm(
     inv = 1.0 / ivy.sqrt(variance + eps)
     if scale is not None:
         inv = inv * scale
-    xnormalized = x * inv.astype(x.dtype, copy=False) + ivy.astype(
-        offset - mean * inv if offset is not None else -mean * inv, x.dtype
+    xnormalized = x * inv.astype(ivy.dtype(x), copy=False) + ivy.astype(
```
Comment (vedpatwardhan) [inline ivy/functional/ivy/experimental/norms.py:198]: ig given that `x` is guaranteed to be an `ivy.Array` we could continue to use `x.dtype` right?

File: ivy/func_wrapper.py [hunk 2]
```diff
@@ -1056,6 +1057,43 @@ def _handle_partial_mixed_function(*args, **kwargs):
...
+            The keyword arguments to be passed to the function.
+
+        Returns
+        -------
+            The return of the function, with the inplace update
+            handled for jax and tensorflow.
+        """
+        normalized, runningmean, runningvar = fn(*args, **kwargs)
+
+        if ivy.curr_backend_str() == "tensorflow":
```
Comment (vedpatwardhan) [inline ivy/func_wrapper.py:1087]: ig this should be `ivy.current_backend_str()`

**Review by vedpatwardhan — CHANGES_REQUESTED** [review]
Hey @hello-fri-end, just requested a couple of changes apart from the debug statements of course. Thanks for diving into this 😄

**Pull request closed without merging**